### PR TITLE
New Article: How to wrap a camel?

### PR DESCRIPTION
**Pull request by G4Vi**
Draft article for https://github.com/perladvent/Perl-Advent/issues/234 .

Comments and revisions welcome!

### CODE REVIEW

**Comment by thibaultduponchelle** [comment]
Some XS 🕺🏼

File: 2022/incoming/how-to-wrap-a-camel-.pod [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        chomp $numhumps;
+        last if($numhumps =~ /^\d+$/);
+        print "Your nose ain't too bright, $numhumps is not a number\n";
+    }
+    my %results = (1 => "Dromedary\n", 2 => "Bactrian\n");
+    print($results{$numhumps} // "Maybe a reindeer? Probably not a camel!\n");
+
+=end perl
+
+Artie noted to Santa that Pure Perl scripts can be wrapped with an existing version of APPerl, off the shelf builds of APPerl can be found at L<https://computoid.com/APPerl/#RELEASES> .
```
Comment (thibaultduponchelle) [inline 2022/incoming/how-to-wrap-a-camel-.pod:40]: I would give a text to this link

**Review by thibaultduponchelle — CHANGES_REQUESTED** [review]

**Comment by thibaultduponchelle** [comment]
Overall sounds good for me. Not tested the code snippets.

I like these kind of "low level" stuff, thank you for contributing this

File: 2022/incoming/how-to-wrap-a-camel-.pod (hunk 1 above)
Comment (G4Vi) [reply 2022/incoming/how-to-wrap-a-camel-.pod]: done in https://github.com/perladvent/Perl-Advent/pull/239/commits/6c8f086089262f2995ac526f205bbdb5deea0f0d

**Comment by thibaultduponchelle** [comment]
It's probably the fault of the pod to html converter but the code snippets have errors when you copy paste them for execution:
```
Unrecognized character \xC2; marked by <-- HERE after <-- HERE near column 1 at code.pl line 4.
```
When I copy from pod, there is no error.

Not critical for me, I'm going to approve

**Review by thibaultduponchelle — APPROVED** [review]

**Comment by G4Vi** [comment]
> It's probably the fault of the pod to html converter but the code snippets have errors when you copy paste them for execution:
> 
> ```
> Unrecognized character \xC2; marked by <-- HERE after <-- HERE near column 1 at code.pl line 4.
> ```
> 
> When I copy from pod, there is no error.
> 
> Not critical for me, I'm going to approve

Weird, I'd really like to get the code snippets working if possible, how did you generate it? I just tried the steps on README (with moving it into articles under a dummy date) and the output HTML only contains ASCII so I'm not sure how the error is possible.

**Comment by thibaultduponchelle** [comment]
Something like:
1. install WWW::AdventCalendar
2. cd 2022
3. move incoming/how-to-wrap-a-camel-.pod to articles/2021-12-15.pod
4. advcal -c advent.ini --today 2022-12-15
5. firefox out/

**Comment by G4Vi** [comment]
Thank you. I found the issue, the html generator replaces leading `\s` with `&nbsp;`. Firefox on Linux (arguably incorrectly) copies `&nbsp;` as `0x20` covering up the problem, but producing usable code. Firefox on Windows copies the `&nbsp;` as the UTF-8 bytes `C2A0`. https://github.com/perladvent/Perl-Advent/issues/263 .

**Comment by thibaultduponchelle** [comment]
In the meantime I think we can move forward with this article.

@openstrike would you be once again available for a review 🙏🏼

File: 2022/incoming/how-to-wrap-a-camel-.pod [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+Artie noted to Santa that Pure Perl scripts can be wrapped with an existing version of APPerl, off the shelf builds of APPerl can be found on the L<APPerl webpage|https://computoid.com/APPerl/#RELEASES>.
+
+Artie picked C<perl-small.com> off the shelf and created an APPerl project:
+
+    wget -O src/perl.com 'https://github.com/G4Vi/Perl-Dist-APPerl/releases/latest/download/perl-small.com'
+    chmod +x src/perl.com
+    apperlm init --name default_config
+
+Artie edited C<default_config> in the newly created C<apperl-project.json> to copy in the C<cameltype> script and set the destination executable name to C<cameltype.com>, the name is important as C<argv[0]> is used to determine what script embedded in APPerl to run.
```
Comment (openstrike) [inline 2022/incoming/how-to-wrap-a-camel-.pod:48]: s/what script/which script/

File: 2022/incoming/how-to-wrap-a-camel-.pod [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    "zip_extra_files" : { "bin" : ["script/merrychristmas"] }
+
+Artie checked out the config, configured, built, and tested out the Merry Christmas message contraption:
+
+    apperlm checkout default_config
+    apperlm configure
+    apperlm build
+    ./merrychristmas.com
+
+Santa said, "Great work, now we can wrap camels of all shapes and sizes to spread cheer to the world! I'd like to put this into practice immediately. Artie replied, "Thank you Santa, I'll get to work teaching all the other elves how to wrap a camel."
```
Comment (openstrike) [inline 2022/incoming/how-to-wrap-a-camel-.pod:129]: Santa's speech is missing hte closing quotation marks after "immediately".

**Review by openstrike — CHANGES_REQUESTED** [review]
This looks really good (and useful!). Just 2 **very** minor tweaks to be looked at and I think it will be ready to go.

File: 2022/incoming/how-to-wrap-a-camel-.pod (hunk 2 above)
Comment (thibaultduponchelle) [inline 2022/incoming/how-to-wrap-a-camel-.pod:48]:
```suggestion
Artie edited C<default_config> in the newly created C<apperl-project.json> to copy in the C<cameltype> script and set the destination executable name to C<cameltype.com>, the name is important as C<argv[0]> is used to determine which script embedded in APPerl to run.
```

File: 2022/incoming/how-to-wrap-a-camel-.pod (hunk 3 above)
Comment (thibaultduponchelle) [inline 2022/incoming/how-to-wrap-a-camel-.pod:129]:
```suggestion
Santa said, "Great work, now we can wrap camels of all shapes and sizes to spread cheer to the world! I'd like to put this into practice immediately". Artie replied, "Thank you Santa, I'll get to work teaching all the other elves how to wrap a camel."
```

**Review by thibaultduponchelle — APPROVED** [review]

**Review by openstrike — APPROVED** [review]
Good to go. Thanks.

**Pull request merged**